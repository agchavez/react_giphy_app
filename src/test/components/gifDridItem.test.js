
import { shallow } from "enzyme";
import React from "react";
import {GifGridItem} from "../../components/GifGridItem";


describe('Pruebas en <GifGidItem />', () => {
    const title = "Este es el titulo";
    const url = "https://localhost/hola.png";
    const wrapper = shallow(<GifGridItem title= {title} url={url}/>)
    

    test('Debe mostrar el componente correctamente ', () => {

        expect(wrapper).toMatchSnapshot();
        
    })

    test('Debe de tener un parafo con el title ', () => {
        const p =  wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe de tener la imagen igual al url y alt de los props ', () => {
        const img = wrapper.find('img');
        const {src, alt } = img.props();
        expect(src).toBe(url);
        expect(alt).toBe(title);

    })

    test('Debe tener la clase animate_fadeIn ', () => {
        const container =  wrapper.find('div');
        const {className} = container.props();
        //Si incluye la palabra 'animate__fadeIn' en lista de clases
        expect(className.includes('animate__fadeIn')).toBe(true)
    })
    
    
    
    
})
