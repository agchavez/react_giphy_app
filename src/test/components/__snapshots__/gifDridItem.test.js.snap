// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGidItem /> Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Este es el titulo"
    src="https://localhost/hola.png"
  />
  <p>
     
    Este es el titulo
     
  </p>
</div>
`;
